dhis-2/dhis-support/dhis-support-cache-invalidation/src/main/java/org/hisp/dhis/cacheinvalidation/redis/PostCacheEventPublisher.java

```diff
@@ -114,6 +162,47 @@ private void publishMessage( Class<?> realClass, String message )
         }
     }
 
+    private Serializable getDataValueId( Object entity )
+    {
+        DataValue dataValue = (DataValue) entity;
+
+        long dataElementId = dataValue.getDataElement().getId();
+        long periodId = dataValue.getPeriod().getId();
+        long organisationUnitId = dataValue.getSource().getId();
+        long categoryOptionComboId = dataValue.getAttributeOptionCombo().getId();
+        long attributeOptionComboId = dataValue.getAttributeOptionCombo().getId();
+
+        return dataElementId + ";" + periodId + ";" + organisationUnitId + ";" + categoryOptionComboId + ";"
```

Comment:
At some point we should think about using dedicated ID classes for this combination of IDs. Joining and splitting is relying on too much implicit knowledge. I guess we could equally well capture these in a small inner class like this? 
```java
@Value
static class DataValueId {
  long dataElementId;
  long periodId;
  long organisationUnitId;
  long categoryOptionComboId;
  long attributeOptionComboId;
}
```
Then you can simply do:
```java
return new DataValueId(dataElementId, periodId, organisationUnitId, categoryOptionComboId, attributeOptionComboId);
```
And so on for all the other IDs you need.

Reply:
The problem here is that I need to serialize the IDs for the message passing through Redis. Without having to serialize and deserialize the messages, just using colon and semicolon separated IDs is by far the easiest.